Remotion/Validation/Core/Validators/MaximumLengthValidator.cs

```diff
@@ -15,18 +15,78 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using JetBrains.Annotations;
+using Remotion.FunctionalProgramming;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
+using Remotion.Validation.Results;
 
 namespace Remotion.Validation.Validators
 {
-  public class MaximumLengthValidator : LengthValidator
+  public class MaximumLengthValidator : ILengthValidator
   {
+    public int Min { get; }
+
+    public int? Max { get; }
+
+    public string ErrorMessage { get; }
+
+    public ValidationMessage ValidationMessage { get; }
+
     public MaximumLengthValidator (int max, [NotNull] ValidationMessage validationMessage)
-        : base(0, max, $"The value must have at most {max} characters.", validationMessage)
+        : this(max, $"The value must have at most {max} characters.", validationMessage)
     {
     }
 
-    // TODO RM-5906: build separate validator with a dedicated message
+    private MaximumLengthValidator (int? max, [NotNull] string errorMessage, [NotNull] ValidationMessage validationMessage)
```

Comment:
We dont need the fluff of the extra overload since they are only ever chained together. Maybe the message could be a field?

Reply:
Yeah, that's actually true. @MichaelKetting What do you think? Should I get rid of the overload here?

Reply:
@KuJo7 There's precedent in other other validator types that work without base-validator classes. please check those and make a suggestion.

Reply:
```suggestion
    public MaximumLengthValidator (int? max, [NotNull] ValidationMessage validationMessage)
    {
      ArgumentUtility.CheckNotNull(nameof(validationMessage), validationMessage);

      if (max != null && max < 0)
        throw new ArgumentOutOfRangeException("max", "Max should be larger than 0.");

      Min = 0;
      Max = max;
      ErrorMessage = $"The value must have at most {max} characters.";
      ValidationMessage = validationMessage;
    }
```

As I checked, I would remove the overload and do it similarly to `ExclusiveRangeValidator` and `InclusiveRangeValidator`